tango/executor.py

```diff
@@ -27,26 +45,39 @@ def __init__(
         self.workspace = workspace
         self.include_package = include_package
 
-    def execute_step(self, step: "Step", is_uncacheable_leaf_step: bool = False) -> None:
+    def execute_step(self, step: "Step") -> None:
         # Import included packages to find registered components.
         if self.include_package is not None:
             for package_name in self.include_package:
                 import_extra_module(package_name)
 
-        if step.cache_results:
-            step.ensure_result(self.workspace)
-        elif is_uncacheable_leaf_step:
-            step.result(self.workspace)
+        step.result(self.workspace)
 
-    def execute_step_graph(self, step_graph: StepGraph, run_name: Optional[str] = None) -> Set[str]:
+    def execute_step_graph(
+        self, step_graph: StepGraph, run_name: Optional[str] = None
+    ) -> ExecutorOutput:
         """
-        Execute a :class:`tango.step_graph.StepGraph`.
+        Execute a :class:`tango.step_graph.StepGraph`. This attempts to execute
+        every step in order. If a step fails, its dependent steps are not run,
+        but unrelated steps are still executed.
         """
 
-        ordered_steps = sorted(step_graph.values(), key=lambda step: step.name)
-        uncacheable_leaf_steps = step_graph.uncacheable_leaf_steps()
+        successful: Set[str] = set()
+        failed: Set[str] = set()
+        not_run: Set[str] = set()
+        error_tracebacks: List[str] = []
+        for step in step_graph.values():
+            if any([dep.name in failed for dep in step.recursive_dependencies]):
```

Comment:
You don't need the extra `[]` there. `any()` can take a generator, and it'll save one list allocation.